src/libs/pluginprocess/pluginprocess.c

```diff
@@ -0,0 +1,379 @@
+ * Also the resulting process information has to be stored for the plugin.
+ * In order to allow users to handle custom plugin data this will not
+ * automatically call elektraPluginSetData.
+ * 
+ * Typically called in a plugin's open function like (assuming no custom plugin data):
+ * @code
+int elektraPluginOpen (Plugin * handle, Key * errorKey)
+{
+	ElektraPluginProcess * pp = elektraPluginGetData (handle);
+	if (pp == NULL)
+	{
+		if ((pp = elektraPluginProcessInit (handle, errorKey)) == NULL) return ELEKTRA_PLUGIN_STATUS_ERROR;
+		elektraPluginSetData (handle, pp);
+		if (!elektraPluginProcessIsParent (pp)) elektraPluginProcessStart (handle, pp);
+	}
+	if (elektraPluginProcessIsParent (pp)) return elektraPluginProcessOpen (pp, errorKey);
+
+	// actual plugin functionality to be executed in a child process
+	return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+}
+ * @endcode
+ *
+ * @param handle the plugin's handle
+ * @param errorKey a key where error messages will be set
+ * @retval NULL if the initialization failed
+ * @retval a pointer to the information
+ * @ingroup processplugin
+ **/
+ElektraPluginProcess * elektraPluginProcessInit (Plugin * handle, Key * errorKey)
+{
+	// First time initialization
+	ElektraPluginProcess * pp;
+	pp = elektraMalloc (sizeof (ElektraPluginProcess));
+	pp->commandPipe = getPipename (handle, "command");
+	pp->resultPipe = getPipename (handle, "result");
+	pp->dump = elektraInvokeOpen ("dump", 0);
+	pp->counter = 0;
+	if (pp->commandPipe && pp->resultPipe && pp->dump)
+	{
+		if (elektraPluginProcessFork (pp, errorKey) < 0)
+		{
+			cleanupPluginData (pp);
+			return NULL;
+		}
+	}
+	else
+	{
+		cleanupPluginData (pp);
+		ELEKTRA_SET_ERROR (190, errorKey, "Failed to initialize the pipenames and the dump plugin");
+		return NULL;
+	}
+	return pp;
+}
+
+/** Call a plugin's open function in a child process
+ *
+ * This will increase the internal counter how often open/close has been called, 
+ * so the opened pipes and forks will not be closed too early.
+ * 
+ * @param pp the data structure containing the plugin's process information
+ * @param errorKey a key where error messages will be set
+ * @retval the return value of the plugin's open function
+ * @see elektraPluginProcessInit for an example how and where this function is typically used
+ * @ingroup processplugin
+ **/
+int elektraPluginProcessOpen (ElektraPluginProcess * pp, Key * errorKey)
+{
+	pp->counter = pp->counter + 1;
+	return elektraPluginProcessSend (pp, ELEKTRA_PLUGIN_OPEN, NULL, errorKey);
+}
+
+/** Cleanup a plugin process
+ *
+ * This will decrease the internal counter how often open/close has been called, 
+ * closing opened pipes when the counter reaches 0. This will not delete the
+ * plugin data associated with the handle as it may contain other data out of
+ * the scope of this library, so this has to be done manually like
+ * @code
+int elektraPluginClose (Plugin * handle, Key * errorKey)
```

Comment:
leftover?

Reply:
No i think not, as its a library and people probably have other data stored in their plugin data handle besides the processplugin information i've left it to the plugin implementor to clear the handle afterwards by setting it to null (or do whatever he wants to do), and this function tells you when the plugin process has ended.